packages/hash/frontend/src/pages/signup.page.tsx

```diff
@@ -204,67 +199,51 @@ const KratosVerificationFlowForm: FunctionComponent = () => {
 const SignupPage: NextPageWithLayout = () => {
   const router = useRouter();
 
-  const {
-    authenticatedUser,
-    kratosSession: _kratosSession,
-    refetch,
-  } = useAuthenticatedUser();
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  const { updateAuthenticatedUser, loading: updateUserLoading } =
+    useUpdateAuthenticatedUser();
 
   const [invitationInfo] = useState<null>(null);
   const [errorMessage, setErrorMessage] = useState<string>();
 
-  const [updateUser, { loading: updateUserLoading }] = useMutation<
-    UpdateUserMutation,
-    UpdateUserMutationVariables
-  >(updateUserMutation, {
-    onCompleted: async ({ updateUser: updatedUser }) => {
-      await refetch();
+  const handleAccountSetupSubmit = async (params: {
+    shortname: string;
+    preferredName: string;
+    responsibility?: string;
+  }) => {
+    const { shortname, preferredName } = params;
+
+    const { updatedAuthenticatedUser, errors } = await updateAuthenticatedUser({
+      shortname,
+      preferredName,
+    });
+
+    if (errors && errors.length > 0) {
```

Comment:
This length check is redundant right?

Reply:
I'm not sure it is in this case, if there's any circumstance where GQL returns `errors: []` then `parseGraphQLError` will throw an error, so am just trying to account for that. I could move this check into that method though.

Reply:
But if `errors: []` then it'd be falsey?

Reply:
No, `[]` is truthy
<img width="114" alt="image" src="https://user-images.githubusercontent.com/42802102/199503668-dbc4c3a4-dde0-4963-be94-b87498dc5d15.png">

Reply:
<img width="107" alt="image" src="https://user-images.githubusercontent.com/25749103/199504159-1c02c96c-bd79-4e5a-aff3-e985f09d9803.png">
right 🙃 

I hate JS.

Reply:
Isn't it still redundant, since you have the exact same check in `useUpdateAuthenticatedUser` before returning `errors`?

Reply:
If we do have the exact same one then should we change the types to make sure its actually captured by TS?

Something like `[Error, ...Error[]]?` or whatever, I'd be wary of us removing it here and then refactoring could mean we don't check it before hand again and the whole `[] is truthy` thing will just be a big pain to discover.